river/preprocessing/ordinal.py

```diff
@@ -0,0 +1,91 @@
+import collections
+import math
+import numbers
+
+import numpy as np
+import pandas as pd
+
+from river import base
+
+
+class OrdinalEncoder(base.MiniBatchTransformer):
+    # INFO: makes sense to add None initially? so it'll be easier to spot
+    def __init__(self, handle_unknown="use_reserved_value", unknown_value=1, encoded_missing_value=0):
+        # self.categories = categories
+        # self.dtype = dtype
+        self.handle_unknown = handle_unknown
+        self.unknown_value = unknown_value
+        self.encoded_missing_value = encoded_missing_value
+        self.categories_ = collections.defaultdict(dict)
+        # self.categories_ = collections.defaultdict(collections.defaultdict)
+
+    def learn_one(self, x):
+        for i, xi in x.items():
+            if self.handle_unknown == "use_reserved_value":
+                # if self.categories_[i].get(xi, None) is None:
+                if xi not in self.categories_[i]:
+                    if xi is None:
+                        self.categories_[i][xi] = self.encoded_missing_value
+                    else:
+                        self.categories_[i][xi] = 2 + len(self.categories_[i])
+        # self.abs_max[i].update(xi)
+
+        return self
+
+    def transform_one(self, x):
+        # return {i: self.categories_[i][xi] for i, xi in x.items()}
+        return {i: self.categories_[i].get(xi, self.unknown_value) for i, xi in x.items()}
+
+    def learn_many(self, X):
+        for col in X.columns:
+            # self.values[col].update(X[col].unique())
+            known_uniqs = np.array(
+                list(self.categories_[col].keys())
+            )  # INFO: convert to numpy array for np.isin to work properly, otherwise unexpected behavior
+            current_uniqs = X[col].unique()
+
+            # if None in known_uniqs:
+            #     current_uniqs.dropna(inplace=True)
+            # elif
+            upd_encoding = dict()
+
+            # current_uniqs = pd.DataFrame(X)['country'].dropna().unique()
+
+            # new_unique_mask = np.isin(current_uniqs, known_uniqs, assume_unique=True, invert=True)
+            new_unique_mask = np.isin(current_uniqs, known_uniqs, assume_unique=True, invert=True)
+
+            new_uniqs = current_uniqs[new_unique_mask]
+            print(f"{known_uniqs=}")
```

Comment:
You can't leave `print` calls

Reply:
Ofc! 
That was intended for debugging. I was going to remove them once we've settled on implementation, hence "WIP" tag (sorry i should have mentioned that)